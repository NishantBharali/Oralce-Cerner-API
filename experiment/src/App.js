import { Routes } from './Routes';
import React from 'react';

export const App = () => {
    return (
        <div className="page-container">
            <Routes />
        </div>
    );
}