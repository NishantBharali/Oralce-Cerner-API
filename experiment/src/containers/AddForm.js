import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { addIdeas } from "../redux/actions/ideaActions";
import FinalFooter from "./Headers and Footers/Footer";
import ComponentHeader from "./Headers and Footers/ComponentHeader";

export const AddForm = () => {

    const dispatch = useDispatch();
    const ideas = useSelector((state) => state.allideas);

    const [ideaTitle, setIdeaTitle] = useState("");
    const [ideaDescription, setIdeaDescription] = useState("");
    const [ideaStorypoints, setIdeaStorypoints] = useState("");


    const history = useHistory();

    let obj = {
                "ideaTitle": ideaTitle,
                "ideaDescription": ideaDescription,
                "ideaStorypoints": ideaStorypoints
            }

            //dispatching the action addIdeas upon click on submit button
            const submit = () => {
                dispatch(addIdeas(obj))
                history.push('/idealisting')
            }
  
    return (
        <>
        <ComponentHeader />
            <form style={{border: '3px solid white', padding: '10px', marginBottom: '50px', marginTop: '50px', backgroundColor: 'transparent', borderRadius: '5px'}}>
                <div className="row justify-content-center">
                    <div className="col-3 center-content">
                        
                        <div className="form-floating mb-3">
                        <h2 style={{textAlign: 'center', color: 'Navy', fontSize: '30px'}}>Add a new Idea</h2>
                        <label className="form-label" style={{height: "24px", width: "231px", color: "white", fontStyle: 'Open Sans', fontSize: "17px", lineHeight: "1px"}}>Title</label>
                            <input id = "input-ideaTitle" style={{boxSizing: 'border-box', height: '39px', width: '426px', border: '2px solid white', borderRadius: '10px', backgroundColor: 'transparent', color: 'white'}} type="text" className="form-control" name="ideaTitle" placeholder=""
                                   value={ideaTitle}
                                   onChange={(e) => setIdeaTitle(e.target.value)}/>
                           
                        </div>
                        
                        <div className="form-floating mb-3">
                        <label className="form-label" style={{height: "24px", width: "231px", color: "white", fontStyle: 'Open Sans', fontSize: "16px", lineHeight: "24px"}}>Description</label>
                        <br />
                            <textarea id = "input-ideaDesc" style={{resize: 'none', boxSizing: 'border-box', height: '100px', width: '426px', border: '2px solid white', borderRadius: '2px', backgroundColor: 'transparent', color: 'white'}} type="text" className="form-control" name="ideaDescription" placeholder=""
                                   value={ideaDescription}
                                   onChange={(e) => setIdeaDescription(e.target.value)}
                            />
                            
                        </div>
                        <div className="form-floating mb-3">
                        <label className="form-label" style={{height: "24px", width: "231px", color: "white", fontStyle: 'Open Sans', fontSize: "16px", lineHeight: "1px"}}>Story points</label>
                            <input id = "input-ideaSP" style={{boxSizing: 'border-box', height: '39px', width: '426px', border: '2px solid white', borderRadius: '10px', backgroundColor: 'transparent', color: 'white'}} type="number" min="1" max="25" className="form-control" name="ideaStorypoints" placeholder=""
                                   value={ideaStorypoints}
                                   onChange={(e) => setIdeaStorypoints(e.target.value)}
                            />
                            
                        </div>
                        <hr/>
                        <button 
                        id="btn-submit"
                        disabled={!ideaTitle || !ideaDescription || !ideaStorypoints}
                        className="ui button" style={{backgroundColor: 'transparent', border: '3px solid white', color: 'floralwhite'}} tableindex="0" onClick={() => submit()}>
                        <div className="visible content">Submit</div>
                       </button>

                            <button 
                        className="ui animated button" style={{backgroundColor: 'transparent', border: '3px solid white', color: 'white', marginTop: '10px'}} tableindex="0" onClick={(e) => history.push('/idealisting')}>
                        <div className="visible content">Back</div>
                        <div className="hidden content">
                            <i className="chevron circle left icon"></i></div></button>

                    </div>
                </div>
            
            </form>
            <FinalFooter />
        </>
    );
};