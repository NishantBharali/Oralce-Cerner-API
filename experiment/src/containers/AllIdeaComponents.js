import React from "react";
import { useSelector } from "react-redux";
import { Link } from "react-router-dom";

const AllIdeaComponents = () => {

    const ideas = useSelector((state) => state.allIdeas.ideas);
    const renderList1 = ideas.map((idea) => {
    const {id, ideaTitle, ideaDescription, ideaStorypoints} = idea;

        return(        
        <div className="four wide column" style={{marginBottom: '50px', marginTop: '60px'}} key={id}>
        
            <Link to={{pathname:`/repository/${id}`, state: {id: id, ideaTitle: ideaTitle, ideaDescription: ideaDescription, ideaStorypoints: ideaStorypoints}}}>
                <div className="ui link cards" style={{height: "100%", maxHeight:"150px"}}>
                    <div className="card" style={{height: "100%", overflowY: "auto", backgroundColor: 'transparent', border: '2px solid white', borderRadius: '10px', borderColor: 'lightyellow'}}>
                        <div className="content">
                            <div className="description" style={{textAlign: "left", font: "small-caption", fontSize: '17px', fontVariant: 'all-petite-caps', color: 'lightgrey'}}><strong>Title: </strong> {ideaTitle}</div>
                            <div className="description" style={{textAlign: "left", font: "small-caption", fontSize: '17px', fontVariant: 'all-petite-caps', color: 'lightgrey'}}><strong>Description: </strong> {ideaDescription}</div>
                            <div className="meta" style={{textAlign: "left", font: "small-caption", fontSize: '17px', fontVariant: 'all-petite-caps', color: 'lightgrey'}}>Story points: {ideaStorypoints}</div>        
                        </div>
                    </div>
                </div>
            </Link>
            
          </div>
          );
    });

    return <>{renderList1}</>;
};

export default AllIdeaComponents;