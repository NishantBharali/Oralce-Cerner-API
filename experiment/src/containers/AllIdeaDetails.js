import React, { useEffect, useState } from "react";
import { useLocation } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { selectedIdeas } from "../redux/actions/ideaActions";
import { useHistory } from "react-router-dom";

const AllIdeaDetails = () => {

  const history = useHistory();
  const dispatch = useDispatch();
  const location = useLocation();

  //extracting the state idea from the global state
  const ideas = useSelector((state) => state.idea);

  const [id, setId] = useState(()=>{return location.state.id});
  const [ideaTitle, setIdeaTitle] = useState(()=>{return location.state.ideaTitle});
  const [ideaDescription, setIdeaDescription] = useState(()=>{return location.state.ideaDescription});
  const [ideaStorypoints, setIdeaStorypoints] = useState(()=>{return location.state.ideaStorypoints});

    //dispatching the action selectedIdeas
    useEffect(() => {
        dispatch(selectedIdeas(id))
    }, [dispatch, id]);

    return(<div className="ui grid container">
    {Object.keys(ideas).length === 0 ? (
      <div>...Loading</div>
    ) : (
      <div className="ui grid container" style={{marginTop: '15px', marginBottom: '5px'}}>
        <div className="ui centered card">
            <div className="content">
              <p style={{textAlign: "left", fontStyle: "-moz-initial", fontVariant: "all-petite-caps"}}><strong>Title: </strong> {ideaTitle}</p>
              <p style={{textAlign: "left", fontStyle: "-moz-initial", fontVariant: "all-petite-caps"}}><strong>Description: </strong> {ideaDescription}</p>
              <p className="meta" style={{textAlign: "left", fontStyle: "-moz-initial", fontVariant: "all-petite-caps"}}><strong>Story points: </strong> {ideaStorypoints}</p>
            </div>
                <button className="ui animated button" tableindex="0" onClick={() => history.push('/repository')}>
                <div className="visible content">Back</div>
                <div className="hidden content">
                <i className="left arrow icon"></i>
                </div>
                </button>
          </div>
        </div>
  
    )}
  </div>
);
};

export default AllIdeaDetails;