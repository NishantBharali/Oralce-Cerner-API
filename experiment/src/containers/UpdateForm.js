import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { useLocation } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { updateIdea } from "../redux/actions/ideaActions";
import FinalFooter from "./Headers and Footers/Footer";
import ComponentHeader from "./Headers and Footers/ComponentHeader";

export const UpdateIdea = () => {

    const location = useLocation();
    const dispatch = useDispatch();

    //extracting the value from the global state
    const ideas = useSelector((state) => state.allIdeas);
    const [id, setId] = useState(()=>{return location.state.id});
    const [ideaTitle, setIdeaTitle] = useState(()=>{return location.state.ideaTitle});
    const [ideaDescription, setIdeaDescription] = useState(()=>{return location.state.ideaDescription});
    const [ideaStorypoints, setIdeaStorypoints] = useState(()=>{return location.state.ideaStorypoints});

    const history = useHistory();


let obj = {
            "id": id,
            "ideaTitle": ideaTitle,
            "ideaDescription": ideaDescription,
            "ideaStorypoints": ideaStorypoints
        }

    //dispatching the action updateIdea upon button click
    const onButtonHandler = () => {
            dispatch(updateIdea(obj))
            history.push('/idealisting')
    }

    return(
   
        <>
        <ComponentHeader />
            <form style={{border: '3px solid white', padding: '20px', marginBottom: '50px', marginTop: '50px', backgroundColor: 'transparent'}}>
                <div className="row justify-content-center">
                    <div className="col-3 center-content">

                        <div className="form-floating mb-3">
                        <h2 style={{textAlign: 'center', color: 'Navy', fontSize: '30px'}}>Update Your Idea</h2>
                        <label className="form-label" style={{color: 'white', fontSize: '15px'}}>Title</label>
                            <input type="text" style={{boxSizing: 'border-box', height: '39px', width: '426px', border: '2px solid white', borderRadius: '10px', backgroundColor: 'transparent', color: 'white'}} className="form-control" name="ideaTitle" placeholder="enter your title"
                                   value={ideaTitle}
                                   onChange={(e) => setIdeaTitle(e.target.value)}/>
                           
                        </div>

                        <div className="form-floating mb-3">
                        <label className="form-label" style={{color: 'white', fontSize: '15px'}}>Description</label>
                            <textarea type="text" style={{resize: 'none', boxSizing: 'border-box', height: '80px', width: '100%', border: '2px solid lightgrey', borderRadius: '10px', backgroundColor: 'transparent', color: 'white'}} className="form-control" name="ideaDescription" placeholder="Description"
                                   value={ideaDescription}
                                   onChange={(e) => setIdeaDescription(e.target.value)}
                            />
                            
                        </div>
                        <div className="form-floating mb-3">
                        <label className="form-label" style={{color: 'white', fontSize: '15px'}}>Story points</label>
                            <input type="number" className="form-control" style={{boxSizing: 'border-box', height: '39px', width: '426px', border: '2px solid white', borderRadius: '10px', backgroundColor: 'transparent', color: 'white'}} name="ideaStorypoints" placeholder="story points"
                                   value={ideaStorypoints}
                                   onChange={(e) => setIdeaStorypoints(e.target.value)}
                            />
                        </div>
                        <hr/>
                        <button 
                        disabled={!ideaTitle || !ideaDescription || !ideaStorypoints}
                        className="ui button" style={{backgroundColor: 'transparent', border: '3px solid white', color: 'lightgrey'}} tableindex="0" onClick={() => onButtonHandler()}>
                        <div className="visible content">Submit</div>
                        </button>

                            <button 
                        className="ui animated button" style={{backgroundColor: 'transparent', border: '3px solid white', color: 'lightgrey', marginTop: '10px'}} tableindex="0" onClick={(e) => history.push('/idealisting')}>
                        <div className="visible content">Back</div>
                        <div className="hidden content">
                            <i className="chevron circle left icon"></i></div></button>



                    </div>
                </div>
            </form>
        <FinalFooter />
        </>
    );
};