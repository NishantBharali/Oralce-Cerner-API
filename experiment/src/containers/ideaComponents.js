import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { deleteIdea } from "../redux/actions/ideaActions";

const IdeaComponents = () => {

    const dispatch = useDispatch();

    //extracting the state from the global state
    const ideas = useSelector((state) => state.allIdeas.ideas);
    const renderList = ideas.map((idea) => {
        const {id, ideaTitle, ideaDescription, ideaStorypoints} = idea;

        //dispatching the action deleteIdea
        const handleDelete = () => {
          if(window.confirm("are you sure you want to delete?")){
            dispatch(deleteIdea(id))        
          }
        }

        return(  
          //pagination here      
        <div className="four wide column" style={{marginBottom: '50px', marginTop: '60px'}} key={id}>
        
        <Link to={{pathname:`/idea/update/${id}`, state: {id: id, ideaTitle: ideaTitle, ideaDescription: ideaDescription, ideaStorypoints: ideaStorypoints}}}>
        <div className="ui link cards" style={{height: "100%", maxHeight:"150px"}}>
          <div className="card" style={{height: "90%", overflowY: "auto", backgroundColor: 'transparent', border: '2px solid white', borderRadius: '10px', borderColor: 'lightblue'}}>
              <div className="content">
                <div className="description" style={{textAlign: "left", font: 'small-caption', fontVariant: 'all-small-caps', fontSize: '17px', color: 'lightgrey'}}><strong>Title: </strong> {ideaTitle}</div>
                <div className="description" style={{textAlign: "left", font: 'small-caption', fontVariant: 'all-small-caps', fontSize: '17px', color: 'lightgrey'}}><strong>Description: </strong> {ideaDescription}</div>
                <div className="meta" style={{textAlign: "left", font: 'small-caption', fontVariant: 'all-small-caps', fontSize: '17px', color: 'lightgrey'}}>Story points: {ideaStorypoints}</div>        
                </div>

              </div>
              </div>
              </Link>
              <br />
              
              <button id="delete-btn" className="ui button" style={{color:"white", backgroundColor: 'transparent'}} tableindex="0" onClick={() => handleDelete()}>
              <div className="hidden content">
                  <i className="trash alternate outline icon"></i></div>
              </button>
              
            
          
          </div>
        
          );
    });

    return <>{renderList}</>;
};

export default IdeaComponents;