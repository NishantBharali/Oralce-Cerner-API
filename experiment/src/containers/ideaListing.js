import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import IdeaComponents from "./ideaComponents";
import { getIdeas } from "../redux/actions/ideaActions";
import FinalHeader from "./Headers and Footers/MainHeader";
import FinalFooter from "./Headers and Footers/Footer";


const IdeaListing = () => {
  

    const dispatch = useDispatch();

    //dispatching getIdeas action
    useEffect(() => {
        dispatch(getIdeas());
               
            }, [dispatch]);

    //extracting state from the global state
    const ideas = useSelector((state) => state);

    

    return(
    <div className="ui grid container">
        <FinalHeader />
    
            {ideas.allIdeas.ideas.length?
        <IdeaComponents />
    : <h2 style={{textAlign: 'center', color: 'lightblue', fontVariant:'all-petite-caps'}}>No data found. Please add a new idea using the Add Idea button using the drop-down!</h2>}

        <FinalFooter />
        
 
    </div>
        
    );
};

export default IdeaListing;