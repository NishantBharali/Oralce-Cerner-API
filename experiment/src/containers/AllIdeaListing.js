import React, {useEffect} from "react";
import { useDispatch, useSelector } from "react-redux";
import { getAllIdeas } from "../redux/actions/ideaActions";
import FinalFooter from "./Headers and Footers/Footer";
import AllIdeaComponents from "./AllIdeaComponents";
import AllIdeaHeader from "./Headers and Footers/AllIdeasHeader";


const AllIdeaListing = () => {
  
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(getAllIdeas());
        
            }, []);

    //extracting the state from the global state
    const ideas = useSelector((state) => state);

    return(<div className="page-container">
        <div className="ui grid container">
    <AllIdeaHeader />
    
            {ideas.allIdeas.ideas.length?
     <AllIdeaComponents />
    : <h2 style={{textAlign: 'center'}}>"No idea found in the repository. Please add your own ideas to view them by going back to your home page."</h2>}
    
 <FinalFooter />
        </div>
           </div>
        
    );
};

export default AllIdeaListing;