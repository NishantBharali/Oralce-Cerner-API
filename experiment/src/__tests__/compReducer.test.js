import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import { configure ,render,shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import "core-js/stable";
import "regenerator-runtime/runtime";
import { createTestStore } from './testStore.test';
import { Provider } from "react-redux";
import { UpdateIdea } from "../containers/UpdateForm";
import AllIdeaDetails from "../containers/AllIdeaDetails";
import App from '../App';

configure({ adapter: new Adapter() });

let store;
describe("Your test1", () => {

    beforeEach(() => {
        store = createTestStore();
    });

    it("check for test cases",()=> {
        const renderer = new ShallowRenderer();
        renderer.render(
            <Provider store={store}>
            <AllIdeaDetails />
            </Provider>
            );
        expect(renderer).toMatchSnapshot();
    })

    it("check for test cases2",()=> {
        const renderer = new ShallowRenderer();
        renderer.render(
            <Provider store={store}>
            <UpdateIdea />
            </Provider>
            );
        expect(renderer).toMatchSnapshot();
    })

    it("check for test cases app",()=> {
        const renderer = new ShallowRenderer();
        renderer.render(
            <Provider store={store}>
            <App />
            </Provider>
            );
        expect(renderer).toMatchSnapshot();
    })

})