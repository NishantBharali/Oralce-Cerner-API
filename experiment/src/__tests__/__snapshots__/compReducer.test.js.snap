// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Your test1 check for test cases 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useDeferredValue": [Function],
    "useEffect": [Function],
    "useId": [Function],
    "useImperativeHandle": [Function],
    "useInsertionEffect": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useResponder": [Function],
    "useState": [Function],
    "useSyncExternalStore": [Function],
    "useTransition": [Function],
  },
  "_element": <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <AllIdeaDetails />
  </Provider>,
  "_firstWorkInProgressHook": Object {
    "memoizedState": Array [
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Object {
          "addNestedSub": [Function],
          "getListeners": [Function],
          "handleChangeWrapper": [Function],
          "isSubscribed": [Function],
          "notifyNestedSubs": [Function],
          "trySubscribe": [Function],
          "tryUnsubscribe": [Function],
        },
      },
      Array [
        Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      ],
    ],
    "next": Object {
      "memoizedState": Array [
        Object {
          "allIdeas": Object {
            "idea": Object {},
            "ideas": Array [],
          },
          "idea": Object {
            "idea": Object {},
            "ideas": Array [],
          },
        },
        Array [
          Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        ],
      ],
      "next": null,
      "queue": null,
    },
    "queue": null,
  },
  "_forcedUpdate": false,
  "_idCounter": 0,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Object {
          "addNestedSub": [Function],
          "getListeners": [Function],
          "handleChangeWrapper": [Function],
          "isSubscribed": [Function],
          "notifyNestedSubs": [Function],
          "trySubscribe": [Function],
          "tryUnsubscribe": [Function],
        },
      }
    }
  >
    <AllIdeaDetails />
  </Context.Provider>,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;

exports[`Your test1 check for test cases app 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useDeferredValue": [Function],
    "useEffect": [Function],
    "useId": [Function],
    "useImperativeHandle": [Function],
    "useInsertionEffect": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useResponder": [Function],
    "useState": [Function],
    "useSyncExternalStore": [Function],
    "useTransition": [Function],
  },
  "_element": <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <UNDEFINED />
  </Provider>,
  "_firstWorkInProgressHook": Object {
    "memoizedState": Array [
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Object {
          "addNestedSub": [Function],
          "getListeners": [Function],
          "handleChangeWrapper": [Function],
          "isSubscribed": [Function],
          "notifyNestedSubs": [Function],
          "trySubscribe": [Function],
          "tryUnsubscribe": [Function],
        },
      },
      Array [
        Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      ],
    ],
    "next": Object {
      "memoizedState": Array [
        Object {
          "allIdeas": Object {
            "idea": Object {},
            "ideas": Array [],
          },
          "idea": Object {
            "idea": Object {},
            "ideas": Array [],
          },
        },
        Array [
          Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        ],
      ],
      "next": null,
      "queue": null,
    },
    "queue": null,
  },
  "_forcedUpdate": false,
  "_idCounter": 0,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Object {
          "addNestedSub": [Function],
          "getListeners": [Function],
          "handleChangeWrapper": [Function],
          "isSubscribed": [Function],
          "notifyNestedSubs": [Function],
          "trySubscribe": [Function],
          "tryUnsubscribe": [Function],
        },
      }
    }
  >
    <UNDEFINED />
  </Context.Provider>,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;

exports[`Your test1 check for test cases2 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useDeferredValue": [Function],
    "useEffect": [Function],
    "useId": [Function],
    "useImperativeHandle": [Function],
    "useInsertionEffect": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useResponder": [Function],
    "useState": [Function],
    "useSyncExternalStore": [Function],
    "useTransition": [Function],
  },
  "_element": <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <UpdateIdea />
  </Provider>,
  "_firstWorkInProgressHook": Object {
    "memoizedState": Array [
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Object {
          "addNestedSub": [Function],
          "getListeners": [Function],
          "handleChangeWrapper": [Function],
          "isSubscribed": [Function],
          "notifyNestedSubs": [Function],
          "trySubscribe": [Function],
          "tryUnsubscribe": [Function],
        },
      },
      Array [
        Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      ],
    ],
    "next": Object {
      "memoizedState": Array [
        Object {
          "allIdeas": Object {
            "idea": Object {},
            "ideas": Array [],
          },
          "idea": Object {
            "idea": Object {},
            "ideas": Array [],
          },
        },
        Array [
          Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        ],
      ],
      "next": null,
      "queue": null,
    },
    "queue": null,
  },
  "_forcedUpdate": false,
  "_idCounter": 0,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Object {
          "addNestedSub": [Function],
          "getListeners": [Function],
          "handleChangeWrapper": [Function],
          "isSubscribed": [Function],
          "notifyNestedSubs": [Function],
          "trySubscribe": [Function],
          "tryUnsubscribe": [Function],
        },
      }
    }
  >
    <UpdateIdea />
  </Context.Provider>,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;
